common/include/pcl/pcl_macros.h

```diff
@@ -67,24 +67,56 @@
 
 #include <pcl/pcl_config.h>
 
+#include <boost/preprocessor/arithmetic/sub.hpp>
+#include <boost/preprocessor/comparison/equal.hpp>
+#include <boost/preprocessor/comparison/less.hpp>
+#include <boost/preprocessor/control/if.hpp>
+#include <boost/preprocessor/stringize.hpp>
+
 // It seems that __has_cpp_attribute doesn't work correctly
 // when compiling with some versions of nvcc so we
 // additionally check if nvcc is used before setting the
-// PCL_DEPRECATED macro to [[deprecated]].
+// PCL_DEPRECATED_IMPL macro to [[deprecated]].
 #if defined(__has_cpp_attribute) && __has_cpp_attribute(deprecated) && !defined(__CUDACC__)
-  #define PCL_DEPRECATED(message) [[deprecated(message)]]
+  #define PCL_DEPRECATED_IMPL(message) [[deprecated(message)]]
 #elif defined(__GNUC__) || defined(__clang__)
-  #define PCL_DEPRECATED(message) __attribute__((deprecated(message)))
+  #define PCL_DEPRECATED_IMPL(message) __attribute__((deprecated(message)))
 #elif defined(_MSC_VER)
   // Until Visual Studio 2013 you had to use __declspec(deprecated).
   // However, we decided to ignore the deprecation for these version because
   // of simplicity reasons. See PR #3634 for the details.
-  #define PCL_DEPRECATED(message)
+  #define PCL_DEPRECATED_IMPL(message)
 #else
-  #warning "You need to implement PCL_DEPRECATED for this compiler"
-  #define PCL_DEPRECATED(message)
+  #warning "You need to implement PCL_DEPRECATED_IMPL for this compiler"
+  #define PCL_DEPRECATED_IMPL(message)
 #endif
 
+#define PCL_DEPRECATED_MODIFY_MSG(major, minor, msg) \
+    "Slated for removal in PCL " \
+    BOOST_PP_STRINGIZE(PCL_MAJOR_VERSION.PCL_MINOR_VERSION) \
+    ": " msg
```

Comment:
The warnings emitted by compilers (at least GCC) have the following structure:

> `LOCATION`: warning: '`SYMBOL`' is deprecated: `MSG` [-Wdeprecated-declarations]

Usually, our `msg` reads like "use xyz() instead", thus in total the emitted output becomes:

```
<source>:25:22: warning: 'int goo()' is deprecated: use xyz() instead [-Wdeprecated-declarations]
```

Are we sure we want to put this "Slated for removal..." message in front of `msg` and not after?